libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m

```diff
@@ -355,68 +315,82 @@ - (void)oauthCoordinator:(SFOAuthCoordinator *)coordinator willBeginBrowserAuthe
     }
     [readWriteLock lock];
     self.authCoordinatorBrowserBlock = callbackBlock;
+    [readWriteLock unlock];
+    
+    if ([self.config.safariViewDelegate respondsToSelector:@selector(authClient:willBeginBrowserAuthentication:)]) {
+        [self.config.safariViewDelegate authClient:self willBeginBrowserAuthentication:callbackBlock];
+    }
+    
     NSString *appName = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"CFBundleDisplayName"];;
     NSString *alertMessage = [NSString stringWithFormat:[SFSDKResourceUtils localizedString:@"authAlertBrowserFlowMessage"], coordinator.credentials.domain, appName];
-
-    if (self.statusAlert) {
-        _statusAlert = nil;
+    
+     __weak typeof(self) weakSelf = self;
+    
+    
+    SFSDKAlertMessage *messageObject = [SFSDKAlertMessage messageWithBlock:^(SFSDKAlertMessageBuilder *builder) {
+        builder.actionOneTitle = [SFSDKResourceUtils localizedString:@"authAlertOkButton"];
+        builder.actionTwoTitle = [SFSDKResourceUtils localizedString:@"authAlertCancelButton"];
+        builder.alertTitle = [SFSDKResourceUtils localizedString:@"authAlertBrowserFlowTitle"];
+        builder.alertMessage = alertMessage;
+        builder.actionOneCompletion = ^{
+            __strong typeof(weakSelf) strongSelf = weakSelf;
+            if ([strongSelf.config.safariViewDelegate respondsToSelector:@selector(authClientDidProceedWithBrowserFlow:)]) {
+                [strongSelf.config.safariViewDelegate authClientDidProceedWithBrowserFlow:strongSelf];
+            }
+            // Let the OAuth coordinator know whether to proceed or not.
+            if (strongSelf.authCoordinatorBrowserBlock) {
+                strongSelf.authCoordinatorBrowserBlock(YES);
+            }
+        };
+        builder.actionTwoCompletion = ^{
+            __strong typeof(weakSelf) strongSelf = weakSelf;
+            if ([strongSelf.config.safariViewDelegate respondsToSelector:@selector(authClientDidCancelBrowserFlow:)]) {
+                [strongSelf.config.safariViewDelegate authClientDidCancelBrowserFlow:strongSelf];
+            }
+
+            // Let the OAuth coordinator know whether to proceed or not.
+            if (strongSelf.authCoordinatorBrowserBlock) {
+                strongSelf.authCoordinatorBrowserBlock(NO);
+            }
+        };
+    }];
+    if ([self.config.delegate respondsToSelector:@selector(authClient:displayMessage:)]){
```

Comment:
displayMessage is not optional, so we don't need to do the `if([... respondsToSelector])`